dayrepo/snippets/static/js/process_add.js

```diff
@@ -1,9 +1,28 @@
-let i = 1 ;
+let limit = 6;
+let bool_select = 8;
+let not_select = 10;
 const addForm = () => {
-  let content_area = document.getElementById("form_area");
-  let clone_element = content_area.cloneNode(true);
-  clone_element.id = i;
-  alert("工程を追加しますか？");
-  content_area.after(clone_element);
-  i++ ;
+  const count = Number(document.getElementById("count").innerHTML);
+  let now_count = count + 1;
+  if (now_count <= limit){
+    alert("工程を追加しますか？");
+    const content_area = document.getElementById("form_area");
+    let clone_element = content_area.cloneNode(true);
+    let content_clone = document.getElementById("add_space");
+    const content_count = content_area.childElementCount;
+    for (let i=0; i<content_count; i++){
+      if (i != bool_select && i <= not_select){
+        clone_element.children[i].children[1].value="";
+      }
+      else if (i == bool_select){
+        clone_element.children[i].children[1].checked="";
+      }
+    }
+    content_clone.before(clone_element)
+    document.getElementById("count").innerHTML = now_count;
+    clone_element.id = now_count;
+  }
+  else {
+    alert("これ以上増やせません");
+  }
```

Comment:
大きい if 文ができた時、早期 return が使えないかどうかは確かめてみるといいよ〜

```suggestion
  if (now_count > limit){
    alert("これ以上増やせません");
    return ;
  }
  
  alert("工程を追加しますか？");
  const content_area = document.getElementById("form_area");
  let clone_element = content_area.cloneNode(true);
  let content_clone = document.getElementById("add_space");
  const content_count = content_area.childElementCount;
  for (let i=0; i<content_count; i++){
    if (i != bool_select && i <= not_select){
      clone_element.children[i].children[1].value="";
    }
    else if (i == bool_select){
      clone_element.children[i].children[1].checked="";
    }
  }
  content_clone.before(clone_element)
  document.getElementById("count").innerHTML = now_count;
  clone_element.id = now_count;
```